sygnal/dataset/base.py

```diff
@@ -94,29 +100,45 @@ def data_gen():
 
 
 class RefGameSamplerBase(Sampler):
+    """Basic sampler that iterates through (shuffled) indicies
+    and draws disctractors without overlap for each sample.
+    """
 
-    def __init__(self, ndistractors):
-        self.k = ndistractors
+    def __init__(self, ndistractors, nsamples, shuffle=False):
+        """Constructor.
+
+        Args:
+            - ndistractors: number of disctractor indicies for each sample
+            - nsamples: total number of samples to be generated
+            - shuffle: wether to shuffle list
+        """
+        self.ndistractors = ndistractors
```

Comment:
This is personal preference, but in literature `k` is used for the number of distractors, so it might make sense to keep it as variable name. Otherwise changing all variable names with `nsomething` to `n_something` or `number_of_something` to be more pythonic in our naming?